core/src/main/java/org/elasticsearch/index/query/MultiMatchQueryBuilder.java

```diff
@@ -793,8 +792,14 @@ protected Query doToQuery(QueryShardContext context) throws IOException {
                 }
             }
         }
-
-        Map<String, Float> newFieldsBoosts = QueryParserHelper.resolveMappingFields(context, fieldsBoosts);
+        Map<String, Float> newFieldsBoosts;
+        if (fieldsBoosts.isEmpty()) {
+            // no fields provided, defaults to index.query.default_field
+            List<String> defaultFields = context.defaultFields();
```

Comment:
Minor nit, we probably don't need to pull this into a variable if we're just immediately passing it into the `parseFieldsAndWeights`